diff --git a/app/src/pointing/input_listener.c b/app/src/pointing/input_listener.c
--- a/app/src/pointing/input_listener.c
+++ b/app/src/pointing/input_listener.c
@@ -191,7 +191,7 @@ static int filter_with_input_config(const struct input_listener_config *cfg,
                 if (ret < 0) {
                     return ret;
                 }
                 if (!override->process_next) {
-                    return 0;
+                    return ret;
                 }
             }
